crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -514,48 +507,32 @@ impl AlgorithmUpdaterV1 {
 
     fn da_block_update(
         &mut self,
-        height_range: Range<u32>,
-        range_cost: u128,
+        heights: &[u32],
+        recording_cost: u128,
     ) -> Result<(), Error> {
-        let expected = self.da_recorded_block_height.saturating_add(1);
-        let first = height_range.start;
-        if first != expected {
-            Err(Error::SkippedDABlock {
-                expected,
-                got: first,
-            })
-        } else {
-            let last = height_range.end.saturating_sub(1);
-            let range_bytes = self.drain_l2_block_bytes_for_range(height_range)?;
-            let new_cost_per_byte: u128 = range_cost.checked_div(range_bytes).ok_or(
-                Error::CouldNotCalculateCostPerByte {
-                    bytes: range_bytes,
-                    cost: range_cost,
-                },
-            )?;
-            self.da_recorded_block_height = last;
-            let new_da_block_cost = self
-                .latest_known_total_da_cost_excess
-                .saturating_add(range_cost);
-            self.latest_known_total_da_cost_excess = new_da_block_cost;
-            self.latest_da_cost_per_byte = new_cost_per_byte;
-            Ok(())
-        }
+        let recorded_bytes = self.drain_l2_block_bytes_for_range(heights)?;
```

Comment:
Maybe just:

```suggestion
        let recorded_bytes = self.drain_l2_block_bytes(heights)?;
```

It'll save us from renaming the function when we change the datatype again. However, the intention here could be to not relate to the `Range<T>`, but to the "range" as arbitrary "noun".

Either way, just a small detail.